feature-find-meal/src/main/java/com/kinandcarta/feature/find/meal/adapter/MealsAdapter.kt

```diff
@@ -1,81 +1,75 @@
+    private val clickListener: (String) -> Unit
 ) : ListAdapter<Meal, MealsAdapter.MealsViewHolder>(MealDiffCallback()) {
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
-        LayoutInflater.from(parent.context)
-            .inflate(R.layout.meal_item_row, parent, false)
-            .run(::MealsViewHolder)
-
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MealsViewHolder {
+       val itemBinding = MealItemRowBinding.inflate(
+           LayoutInflater.from(parent.context), parent, false)
+        return MealsViewHolder(itemBinding)
+    }
 
     override fun onBindViewHolder(holder: MealsViewHolder, position: Int) {
         val item = getItem(position)
         holder.bindData(item, position)
     }
 
-    inner class MealsViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
-        private val nameView = itemView.findViewById<TextView>(R.id.mealName)
-        private val tempView = itemView.findViewById<TextView>(R.id.mealTemp)
-        private val infoView = itemView.findViewById<TextView>(R.id.mealInfo)
-        private val distanceView = itemView.findViewById<TextView>(R.id.mealDistance)
-        private val availableView = itemView.findViewById<TextView>(R.id.mealAvailable)
-        private val expiryView = itemView.findViewById<TextView>(R.id.mealExpiry)
-        private val portionsView = itemView.findViewById<TextView>(R.id.mealPortions)
-        private val reserveButton = itemView.findViewById<Button>(R.id.reserveButton)
+    inner class MealsViewHolder(private val itemBinding: MealItemRowBinding): RecyclerView.ViewHolder(itemBinding.root) {
+        private val ctx = itemBinding.root.context
 
         fun bindData(meal: Meal, position: Int) {
-            nameView.text = meal.name
-            val tempString = if (meal.hot) "Hot" else "Cold"
-            val tempColor = if (meal.hot) R.color.colorHot else R.color.colorCold
-            tempView.text = tempString
-            tempView.setTextColor(ContextCompat.getColor(itemView.context, tempColor))
+            itemBinding.mealName.text = meal.name
+            setupMealTemp(meal.hot)
+            setupMealInfo(meal.info)
+            itemBinding.mealDistance.text = "${meal.distance} ${distanceUnit}"
+            itemBinding.mealAvailable.text = "${ctx.getString(R.string.available_leader)} ${meal.availableFromDate}"
+            itemBinding.mealExpiry.text = "${ctx.getString(R.string.expiry_leader)} ${meal.expiryDate}"
+            itemBinding.mealPortions.text = "${meal.quantity.getPortionsString()}"
+            setupReserveButton(meal.quantity, meal.id)
+        }
 
-            if (meal.info.isNullOrEmpty()) {
-                infoView.isVisible = false
+        private fun setupMealTemp(isHot: Boolean) {
+            val tempString = if (isHot) ctx.getString(R.string.hot) else ctx.getString(R.string.cold)
+            val tempColor = if (isHot) R.color.colorHot else R.color.colorCold
```

Comment:
I personally don't like adding logic in ViewHolders, but I'd suggest this as a future improvement and not a blocker for merging